proposals/per_policy_webhook_configurations.md

```diff
@@ -0,0 +1,235 @@
+# Meta
+[meta]: #meta
+- Name: Policy-based Webhook Configurations
+- Start Date: 2023-12-12
+- Update data (optional):
+- Author(s): @realshuting
+
+# Table of Contents
+[table-of-contents]: #table-of-contents
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [Implementation](#implementation)
+- [Migration (OPTIONAL)](#migration-optional)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Prior Art](#prior-art)
+- [Unresolved Questions](#unresolved-questions)
+- [CRD Changes (OPTIONAL)](#crd-changes-optional)
+
+# Overview
+[overview]: #overview
+
+Configure Kyverno resource webhook configurations based on policy settings.
+
+# Definitions
+[definitions]: #definitions
+
+Kyverno has two webhook configurations registered for the selected resources by policies, a MutatingWebhookConfiguration and a ValidatingWebhookConfiguration respectively. For brevity, this proposal simply uses "webhooks" to refer to the two mentioned webhook configurations.
```

Comment:
Are you also only talking about the `-resource` webhook for each? There are different webhooks for different purposes.

Reply:
I believe it will be for `-resource` webhooks only.

Reply:
Yes, for resource webhooks only.